tests/integration/testlib/retry.py

```diff
@@ -0,0 +1,31 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+import time
+
+
+def retry_till_true_or_timeout(timeout, func, *args, **kwargs):
```

Comment:
I think the the "stop" condition can be documented and not added to the function name.
It should be obvious that one of the retry stop condition is the timeout if that is provided.

Regarding the `func` returning true, that may be a bit harder to gasp, but I think it is pretty straight forward as well (otherwise, why would one use a retry at all).
If you want to be more modular and specific, you can add a `success_func_condition` function as input and default it to `lambda ret: bool(ret)`.

Reply:
This is a test only function and I think current name is good.

Reply:
I personally do not like the name, but I won't pursue it.